src/sink/ws.rs

```diff
@@ -320,8 +324,23 @@ async fn ws_connection_loop(
                                 correlation.as_ref(),
                             )
                             .await?;
-                            // close connection explicitly
-                            ws_stream.close(None).await?;
+                            if !matches!(
```

Comment:
could up extract the matches into a function so we only need to update it in one place?

Reply:
done